File: src/main/java/com/FoxThePrezident/entities/templates/Potion.java
package com.FoxThePrezident.entities.templates;

import com.FoxThePrezident.Data;
import com.FoxThePrezident.Debug;
import com.FoxThePrezident.entities.Item;
import com.FoxThePrezident.entities.player.Player;
import com.FoxThePrezident.listeners.Listeners;
import com.FoxThePrezident.listeners.RefreshListener;
import com.FoxThePrezident.graphics.Graphics;
import com.FoxThePrezident.graphics.Icons;

import javax.swing.*;

/**
 * Interactive potion.
 */
public class Potion implements RefreshListener {
	protected final int[] position;
	protected ImageIcon icon;
	private final Graphics graphics = new Graphics();
	private final Listeners listeners = new Listeners();

	protected int heal = 0;

	public Potion(int[] Position) {
		if (Debug.entities.templates.Potion) System.out.println("--- [Potion.constructor]");
		position = Position;
	}

	@Override
	public int[] getPosition() {
		if (Debug.entities.templates.Potion) System.out.println("--- [Potion.getPosition]");
		return position;
	}

	@Override
	public void onRefresh() {
		if (Debug.entities.templates.Potion) System.out.println(">>> [Potion.onRefresh]");

		if (Data.running) {
			if (position[0] == Data.Player.position[0] && position[1] == Data.Player.position[1]) {
				Item item = new Item(Icons.Interactive.hp_potion);
				item.setHeal(heal);
				Player.addItem(item);
				listeners.removeRefreshListener(this);
				if (Debug.entities.templates.Potion) System.out.println("<<< [Potion.onRefresh] Premature exit due already being used");
				return;
			}
		}

		graphics.drawTile(position, icon, graphics.DECOR_LAYER);
		if (Debug.entities.templates.Potion) System.out.println("<<< [Potion.onRefresh]");
	}
}

File: composeApp/src/commonMain/kotlin/com/one_of_many_simons/one_button_game/entities/templates/Potion.kt
package com.one_of_many_simons.one_button_game.entities.templates

import androidx.compose.ui.graphics.ImageBitmap
import com.one_of_many_simons.one_button_game.Data
import com.one_of_many_simons.one_button_game.Debug.Flags.Entities.Templates.POTION
import com.one_of_many_simons.one_button_game.Debug.Levels.CORE
import com.one_of_many_simons.one_button_game.Debug.Levels.INFORMATION
import com.one_of_many_simons.one_button_game.Debug.debug
import com.one_of_many_simons.one_button_game.Libraries.graphics
import com.one_of_many_simons.one_button_game.Libraries.listeners
import com.one_of_many_simons.one_button_game.dataClasses.Position
import com.one_of_many_simons.one_button_game.entities.player.Item
import com.one_of_many_simons.one_button_game.entities.player.Player
import com.one_of_many_simons.one_button_game.graphics.Graphics.Companion.ENTITIES_LAYER
import com.one_of_many_simons.one_button_game.graphics.Icons
import com.one_of_many_simons.one_button_game.listeners.RefreshListener

/**
 * Interactive potion.
 */
open class Potion(position: Position) : RefreshListener {
    @JvmField
    var position: Position = Position()

    @JvmField
    protected var icon: ImageBitmap? = null

    @JvmField
    protected var heal: Int = 0

    init {
        debug(POTION, CORE, "--- [Potion.constructor]")

        this.position = Position(position)
    }

    override fun onRefresh() {
        debug(POTION, CORE, ">>> [Potion.onRefresh]")

        if (Data.running) {
            if (Data.Player.position.equals(position)) {
                val item = Item(Icons.Interactive.hp_potion)
                item.setHeal(heal)
                Player.addItem(item)
                listeners.removeRefreshListener(this)

                debug(POTION, CORE, "<<< [Potion.onRefresh] Premature exit due already being used")
                return
            }
        }

        graphics.drawTile(position, icon, ENTITIES_LAYER)

        debug(POTION, CORE, "<<< [Potion.onRefresh]")
    }

    override fun getPosition(): Position {
        debug(POTION, INFORMATION, "--- [Potion.getPosition]")

        return position
    }
}
